unit_tests/turbulence/test_turbulence_LES.cpp

```diff
@@ -0,0 +1,272 @@
+                const amrex::Real y = problo[1] + (j + offset) * dx[1];
+                const amrex::Real z = problo[2] + (k + offset) * dx[2];
+
+                farr(i, j, k, 0) = x / sqrt(6.0) * srate;
+                farr(i, j, k, 1) = y / sqrt(6.0) * srate;
+                farr(i, j, k, 2) = z / sqrt(6.0) * srate;
+            });
+        }
+    }
+}
+
+void init_field1(amr_wind::Field& fld, amrex::Real tgrad)
+{
+    const auto& mesh = fld.repo().mesh();
+    const int nlevels = fld.repo().num_active_levels();
+
+    amrex::Real offset = 0.0;
+    if (fld.field_location() == amr_wind::FieldLoc::CELL) {
+        offset = 0.5;
+    }
+
+    for (int lev = 0; lev < nlevels; ++lev) {
+        const auto& dx = mesh.Geom(lev).CellSizeArray();
+        const auto& problo = mesh.Geom(lev).ProbLoArray();
+
+        for (amrex::MFIter mfi(fld(lev)); mfi.isValid(); ++mfi) {
+            auto bx = mfi.growntilebox();
+            const auto& farr = fld(lev).array(mfi);
+
+            amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+                const amrex::Real z = problo[2] + (k + offset) * dx[2];
+
+                farr(i, j, k, 0) = z * tgrad;
+            });
+        }
+    }
+}
+
+} // namespace
+
+class TurbLESTest : public MeshTest
+{
+protected:
+    void populate_parameters() override
+    {
+        MeshTest::populate_parameters();
+
+        {
+            amrex::ParmParse pp("amr");
+            amrex::Vector<int> ncell{{10, 20, 30}};
+            pp.addarr("n_cell", ncell);
+            pp.add("blocking_factor", 2);
+        }
+        {
+            amrex::ParmParse pp("geometry");
+            amrex::Vector<amrex::Real> problo{{0.0, 0.0, 0.0}};
+            amrex::Vector<amrex::Real> probhi{{10.0, 10.0, 10.0}};
+            pp.addarr("prob_lo", problo);
+            pp.addarr("prob_hi", probhi);
+        }
+    }
+
+    const amrex::Real dx = 10.0 / 10.0;
+    const amrex::Real dy = 10.0 / 20.0;
+    const amrex::Real dz = 10.0 / 30.0;
+};
+
+TEST_F(TurbLESTest, test_smag_setup_calc)
+{
+    // Parser inputs for turbulence model
+    const amrex::Real Cs = 0.16;
+    const amrex::Real visc = 1e-5;
+    {
+        amrex::ParmParse pp("turbulence");
+        pp.add("model", (std::string) "Smagorinsky");
+    }
+    {
+        amrex::ParmParse pp("Smagorinsky_coeffs");
+        pp.add("Cs", Cs);
+    }
+    {
+        amrex::ParmParse pp("transport");
+        pp.add("viscosity", visc);
+    }
+
+    // Initialize necessary parts of solver
+    populate_parameters();
+    initialize_mesh();
+    auto& pde_mgr = sim().pde_manager();
+    pde_mgr.register_icns();
+    sim().init_physics();
+
+    // Create turbulence model
+    sim().create_turbulence_model();
+    // Get turbulence model
+    auto& tmodel = sim().turbulence_model();
+
+    // Get coefficients
+    auto model_dict = tmodel.model_coeffs();
+
+    for (const std::pair<const std::string, const amrex::Real> n : model_dict) {
```

Comment:
Not sure why there are two constants

Reply:
there's only one constant. n.first is the name of the constant, and n.second is the value of the constant.